sdk/src/account.rs

```diff
@@ -77,8 +77,32 @@ impl From<Account> for AccountSharedData {
     }
 }
 
+#[derive(Debug)]
+pub enum LamportsError {
```

Comment:
@jackcmay would it make sense for this to live in the `solana-program` crate so we can implement `From<LamportsError> for InstructionError` and get rid of some of the explicit `map_err()` calls

Reply:
Or we create InstructionErrors for these and return those directly?

Reply:
I was thinking they might be useful in places where we don't currently return `InstructionError`

Reply:
I just made up an error because I wasn't sure what you guys may prefer. Please tell me what you prefer.

Reply:
What @t-nelson said :-)